c++/src/kj/common-test.c++

```diff
@@ -940,5 +940,15 @@ KJ_TEST("ArrayPtr::as<Std>") {
   KJ_EXPECT(stdPtr.size() == 5);
 }
 
+// Verifies the expected values of kj::isDisallowedInCoroutine<T>
+
+struct DisallowedInCoroutine {
+  KJ_DISALLOW_AS_COROUTINE_PARAM;
+};
+struct AllowedInCoroutine {};
+
+static_assert(_::isDisallowedInCoroutine<DisallowedInCoroutine>());
+static_assert(!_::isDisallowedInCoroutine<AllowedInCoroutine>());
```

Comment:
Maybe add tests for reference and pointer types?

Reply:
Ah right, good call.